sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/MapElementsTest.java

```diff
@@ -118,12 +121,41 @@ public void testVoidValues() throws Exception {
     pipeline.run();
   }
 
+  @Test
+  public void testSerializableFunctionDisplayData() {
+    SerializableFunction<Integer, Integer> serializableFn =
+        new SerializableFunction<Integer, Integer>() {
+          @Override
+          public Integer apply(Integer input) {
+            return input;
+          }
+        };
+
+    MapElements<?, ?> serializableMap = MapElements.via(serializableFn)
+        .withOutputType(TypeDescriptor.of(Integer.class));
+    assertThat(DisplayData.from(serializableMap),
+        hasDisplayItem("mapFn", serializableFn.getClass()));
+  }
+
+  @Test
+  public void testSimpleFunctionDisplayData() {
+    SimpleFunction<?, ?> simpleFn = new SimpleFunction<Integer, Integer>() {
+      @Override
+      public Integer apply(Integer input) {
+        return input;
+      }
+    };
+
+    MapElements<?, ?> simpleMap = MapElements.via(simpleFn);
+    assertThat(DisplayData.from(simpleMap), hasDisplayItem("mapFn", simpleFn.getClass()));
+  }
+
   static class VoidValues<K, V>
       extends PTransform<PCollection<KV<K, V>>, PCollection<KV<K, Void>>> {
 
     @Override
     public PCollection<KV<K, Void>> apply(PCollection<KV<K, V>> input) {
-      return input.apply(MapElements.<KV<K, V>, KV<K, Void>>via(
+      return input.apply(MapElements.via(
```

Comment:
Make sure you've tested this with a JDK 7 -- sometimes these are necessary for older version support.